okhttp/src/jvmMain/kotlin/okhttp3/internal/cache/CacheStrategy.kt

```diff
@@ -274,7 +274,7 @@ class CacheStrategy internal constructor(
 
       val responseDuration = receivedResponseMillis - sentRequestMillis
       val residentDuration = nowMillis - receivedResponseMillis
-      return receivedAge + responseDuration + residentDuration
+      return maxOf(0, receivedAge + responseDuration + residentDuration)
```

Comment:
I think it’s more straightforward to clamp each of the durations to be non-negative instead of doing that to the overall result.

Reply:
Thanks for your review, I have recommitted the file